src/main/assets/js/local-authorities.ts

```diff
@@ -0,0 +1,132 @@
+import $ from 'jquery';
+import {AjaxErrorHandler} from './ajaxErrorHandler';
+
+
+const { initAll } = require('govuk-frontend');
+
+export class LocalAuthoritiesController {
+  private formId = '#localAuthoritiesForm';
+  private tabId = '#localAuthoritiesTab';
+  private localAuthoritiesContentId = '#localAuthoritiesContent';
+  private localAuthoritiesTabId ='#tab_local-authorities';
+  private loccalAuthoritiesListId = '#localAuthoritiesList';
+
+  private slug = $('#slug').val();
+  private areaOfLaw = ($('#courtAreasOfLaw').val() == undefined ) ? 'unknown': $('#courtAreasOfLaw').val();
+
+  private getLocalAuthoritiesSelect = '#courtAreasOfLaw';
+
+
+  constructor() {
+    this.initialize();
+
+  }
+
+  private initialize(): void {
+    $(() => {
+      if ($(this.tabId).length > 0) {
+        this.displayLocalAuthoritiesTab();
+        this.getAreasOfLaw();
+        this.setUpGetLocalAuthoritiesEventHandler();
+        this.setUpSubmitEventHandler();
+      }
+    });
+  }
+
+
+  private async updateContent(content: any , contentId: string): Promise<void> {
+
+    $(contentId).html(content);
+    initAll({scope: document.getElementById('localAuthoritiesTab')});
+
+  }
+
+  private getAreasOfLaw(): void{
+    $.ajax({
+      url: `/courts/${this.slug}/local-authorities-areas-of-law`,
+      method: 'get',
+      success: async (res) => {
+        await this.updateContent(res, this.localAuthoritiesContentId);
+        this.disableLocalAuthoritiesTab();
+      },
+      error: (jqxhr, errorTextStatus, err) =>
+        AjaxErrorHandler.handleError(jqxhr, 'GET local authorities areas of law failed.')
+    });
+
+  }
+
+  private getLocalAuthorities(): void{
+    $.ajax({
+      url: `/courts/${this.slug}/${this.areaOfLaw}/local-authorities`,
+      method: 'get',
+      success: async (res) => {
+        await this.updateContent(res, this.localAuthoritiesContentId);
+        this.scrollable();
+
+      },
+      error: (jqxhr, errorTextStatus, err) =>
+        AjaxErrorHandler.handleError(jqxhr, 'GET local authorities failed.')
+    });
+
+  }
+
+  private setUpSubmitEventHandler(): void {
+    $(this.formId).on('submit', e => {
+      e.preventDefault();
+      this.areaOfLaw = $('#courtAreasOfLaw').val();
+      const url = `/courts/${this.slug}/${this.areaOfLaw}/local-authorities`;
+      $.ajax({
+        url: url,
+        method: 'put',
+        data: $(e.target).serialize()
+      }).done( async res => {
+        await this.updateContent(res, this.localAuthoritiesContentId);
+        window.scrollTo(0, 0);
+        this.scrollable();
+      }).fail(response =>
+        AjaxErrorHandler.handleError(response, 'POST local authorities failed.'));
+    });
+  }
+
+  private displayLocalAuthoritiesTab(): void{
+
+    const isSuperAdmin = Boolean($('#isSuperAdmin').val() === 'true');
+    if(!isSuperAdmin){
```

Comment:
I think we use middleware to check if the user is a super admin or not, before the tab is displayed? 

```
  app.get('/courts/:slug/postcodes', isSuperAdmin, app.locals.container.cradle.postcodesController.get);
  app.post('/courts/:slug/postcodes', isSuperAdmin, app.locals.container.cradle.postcodesController.post);
```

 Guessing that maybe its the case that we put this here as there a subset of the page that requires super admin rights? If so then not a problem :-)

Reply:
I have removed the logic to check SuperAdmin from the local-authorities.ts to the edit-court-general.njk using the IsSuperAdmin provided by the middleware